File: _jump_game.ts
function canJump(nums: number[]): boolean {
    let goal = nums.length -1;

    for(let i = nums.length-2; i>=0; i--)
    {
        if(nums[i]+i>=goal){
            goal=i
        }
    }

    if(goal==0){
        return true
    }
    else{
        return false;
    }


};

File: Python/Jump_game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:

        res = [False] * (len(nums))
        res[len(nums)-1] = True

        pos = len(nums)-1
        
        for i in range(len(nums)-2,-1, -1):
            steps = nums[i]
            if(i+steps>=pos):
                res[i] = True
                pos = i


        return res[0]
